src/contracts/atlas/Escrow.sol

```diff
@@ -430,57 +422,149 @@ abstract contract Escrow is AtlETH {
     /// @param dAppReturnData Data returned from the execution of the associated UserOperation, which may be required
     /// for the SolverOperation's logic.
-    /// @param lockBytes The packed bytes form of the current EscrowKey state.
-    /// @return A SolverOutcome enum value encoded as a uint256 bitmap, representing the result of the SolverOperation
+    /// @return result SolverOutcome enum value encoded as a uint256 bitmap, representing the result of the
+    /// SolverOperation
+    /// @return solverTracker Tracking data for the solver's bid
     function _solverOpWrapper(
         uint256 bidAmount,
         uint256 gasLimit,
-        address environment,
+        EscrowKey memory key,
         SolverOperation calldata solverOp,
-        bytes memory dAppReturnData,
-        bytes memory lockBytes
+        bytes memory dAppReturnData
     )
         internal
-        returns (uint256)
+        returns (uint256 result, SolverTracker memory solverTracker)
+    {
+        bool success;
+
+        // Calls the solverCall function, just below this function, which will handle calling solverPreTryCatch and
+        // solverPostTryCatch via the ExecutionEnvironment, and in between those two hooks, the actual solver call
+        // directly from Atlas to the solver contract (not via the ExecutionEnvironment).
+        bytes memory data = abi.encodeCall(IAtlas.solverCall, (bidAmount, gasLimit, key, solverOp, dAppReturnData));
+        (success, data) = address(this).call(data);
+
+        if (success) {
+            // If solverCall() was successful, intentionally leave result unset as 0 indicates success
+            solverTracker = abi.decode(data, (SolverTracker));
+        } else {
+            // If solverCall() failed, catch the error and encode the failure case in the result uint accordingly.
+            bytes4 errorSwitch = bytes4(data);
+            if (errorSwitch == AlteredControl.selector) {
+                result = 1 << uint256(SolverOutcome.AlteredControl);
+            } else if (errorSwitch == InsufficientEscrow.selector) {
+                result = 1 << uint256(SolverOutcome.InsufficientEscrow);
+            } else if (errorSwitch == PreSolverFailed.selector) {
+                result = 1 << uint256(SolverOutcome.PreSolverFailed);
+            } else if (errorSwitch == SolverOperationReverted.selector) {
+                result = 1 << uint256(SolverOutcome.SolverOpReverted);
+            } else if (errorSwitch == PostSolverFailed.selector) {
+                result = 1 << uint256(SolverOutcome.PostSolverFailed);
+            } else if (errorSwitch == IntentUnfulfilled.selector) {
+                result = 1 << uint256(SolverOutcome.IntentUnfulfilled);
+            } else if (errorSwitch == SolverBidUnpaid.selector) {
+                result = 1 << uint256(SolverOutcome.BidNotPaid);
+            } else if (errorSwitch == BalanceNotReconciled.selector) {
```

Comment:
Error and SolverOutcome names differ, suggest renaming the error:

```diff
-    error SolverBidUnpaid();
+    error BidNotPaid();
```

Reply:
Fixed here https://github.com/FastLane-Labs/atlas/pull/225/commits/1632ec2221a5641fe207b35b9085fb8bb0c8a419